var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -71,10 +71,17 @@ class Gcc(AutotoolsPackage):
 
     # https://gcc.gnu.org/install/prerequisites.html
     depends_on('gmp@4.3.2:')
-    depends_on('mpfr@2.4.2:')
+    # GCC 7.3 does not compile with newer releases on some platforms, see
+    #   https://github.com/spack/spack/issues/6902#issuecomment-433030376
+    depends_on('mpfr@2.4.2:3.1.6')
     depends_on('mpc@0.8.1:', when='@4.5:')
-    depends_on('isl@0.14', when='@5:5.9')
-    depends_on('isl@0.15:', when='@6:')
+    # Already released GCC versions do not support any newer version of ISL
+    #   GCC 5.4 https://github.com/spack/spack/issues/6902#issuecomment-433072097
+    #   GCC 7.3 https://github.com/spack/spack/issues/6902#issuecomment-433030376
+    #   GCC 9+  https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86724
+    depends_on('isl@0.15', when='@5:5.9')
+    depends_on('isl@0.15:0.18', when='@6:8.9')
```

Comment:
Given that `8.2.0` is the current latest release, it is possible that `@8.3.0:` will work with `isl@20`, so it may be safer to do something like:

`depends_on('isl@0.15:0.18', when='@6:8.2')`

Reply:
It's possible but will not add any performance benefits that one would expect. Therefore, I would prefer stability of use for our users over speculative larger supported range which does not buy our users anything but fancyness alpha testing of potentially breaking releases.